salesforce/salesforce-inbound-endpoint/salesforce-inbound-endpoint-1.0.0/src/main/java/org/wso2/carbon/inbound/salesforce/poll/SalesforceStreamData.java

```diff
@@ -0,0 +1,323 @@
+    protected static String loginEndpoint;
+
+    private String userName;
+    private String password;
+    private String salesforceObject;
+    private String channel;
+    private String packageVersion;
+    private String packageName;
+    private String streamingEndpointUri;
+    private String injectingSeq;
+
+    // optional parameters.
+    private static int connectionTimeout;
+    private static int readTimeout;
+    private static int waitTime;
+
+    private BayeuxClient client = null;
+    private HttpClient httpClient;
+
+    private boolean isPolled = false;
+
+    public SalesforceStreamData(Properties salesforceProperties, String name,
+                                SynapseEnvironment synapseEnvironment, long scanInterval,
+                                String injectingSeq, String onErrorSeq, boolean coordination,
+                                boolean sequential) {
+        super(salesforceProperties, name, synapseEnvironment, scanInterval,
+                injectingSeq, onErrorSeq, coordination, sequential);
+        loadMandatoryParameters(salesforceProperties);
+        loadOptionalParameters(salesforceProperties);
+        streamingEndpointUri = version ? "/" + packageName : "/" + packageName + "/" + packageVersion;
+        channel = version ? "/" + salesforceObject : "/topic/" + salesforceObject;
+        this.injectingSeq = injectingSeq;
+        httpClient = new HttpClient();
+    }
+
+    private void makeConnect() {
+        try {
+            client = makeClient();
+        } catch (Exception e) {
+            handleException("Error during make the client: " + e.getMessage(), e);
+        }
+
+        if (client != null) {
+            client.getChannel(Channel.META_HANDSHAKE).addListener(new ClientSessionChannel.MessageListener() {
+                public void onMessage(ClientSessionChannel channel, Message message) {
+                    log.info("[CHANNEL:META_HANDSHAKE with the Channel for Salesforce Streaming Endpoint]: " + message);
+                    boolean success = message.isSuccessful();
+                    if (!success) {
+                        String error = (String) message.get("error");
+                        if (StringUtils.isNotEmpty(error)) {
+                            handleException("Error during HANDSHAKE: " + error);
+                        }
+                        Exception exception = (Exception) message.get("exception");
+                        if (exception != null) {
+                            handleException("Exception during HANDSHAKE: ", exception);
+                        }
+                    }
+                }
+            });
+
+            client.getChannel(Channel.META_CONNECT).addListener(new ClientSessionChannel.MessageListener() {
+                public void onMessage(ClientSessionChannel channel, Message message) {
+                    log.info("[CHANNEL:META_CONNECT with the Channel for Salesforce Streaming Endpoint]: " + message);
```

Comment:
Just one log in the happy path is enough. Move these to debug level. Need to log in case of errors and when retrying.